packages/embark/src/cmd/cmd_controller.js

```diff
@@ -235,15 +238,19 @@ class EmbarkController {
 
           engine.events.request("webserver:start")
 
-          let contractsFiles = await engine.events.request2("config:contractsFiles");
-          let compiledContracts = await engine.events.request2("compiler:contracts:compile", contractsFiles);
-          let _contractsConfig = await engine.events.request2("config:contractsConfig");
-          let contractsConfig = cloneDeep(_contractsConfig);
-          let [contractsList, contractDependencies] = await engine.events.request2("contracts:build", contractsConfig, compiledContracts);
-          await engine.events.request2("deployment:contracts:deploy", contractsList, contractDependencies);
-          console.dir("deployment done")
+          try {
+            let contractsFiles = await engine.events.request2("config:contractsFiles");
+            let compiledContracts = await engine.events.request2("compiler:contracts:compile", contractsFiles);
+            let _contractsConfig = await engine.events.request2("config:contractsConfig");
+            let contractsConfig = cloneDeep(_contractsConfig);
+            let [contractsList, contractDependencies] = await engine.events.request2("contracts:build", contractsConfig, compiledContracts);
+            await engine.events.request2("deployment:contracts:deploy", contractsList, contractDependencies);
+            console.dir("deployment done")
 
-          await engine.events.request2("watcher:start")
+            await engine.events.request2("watcher:start")
+          } catch (e) {
+            console.log(e);
```

Comment:
We should probably pass the error to the callback. Also, if we do want to log the error here, it should probably be `this.logger.error` or `engine.logger.error`.

Reply:
The callback is already called at the top, it's not actually dependent on the completion of the deployment.
It's a bit weird, but it works.
I'll change it to a logger call though